handler/rfc8628/strategy_hmacsha.go

```diff
@@ -18,6 +20,83 @@ import (
 	enigma "github.com/ory/fosite/token/hmac"
 )
 
+// DeviceFlowSession is a fosite.Session container specific for the device flow.
+type DeviceFlowSession interface {
```

Comment:
This is the specific session for device flow to use. It allows the clients to set the flag when the user completes the browser flow. Therefore, we can handle the `Authorization Pending` status if the flag is not set when handling the access token request.